charts/karpenter/values.yaml

```diff
@@ -72,14 +72,20 @@ affinity:
         - matchExpressions:
             - key: karpenter.sh/provisioner-name
               operator: DoesNotExist
+  podAntiAffinity:
+    preferredDuringSchedulingIgnoredDuringExecution:
+      - labelSelector:
+          matchExpressions:
+            - key: app.kubernetes.io/name
+              operator: In
+              values:
+              - karpenter
```

Comment:
Does this always match with: `app.kubernetes.io/name: {{ include "karpenter.name" . }}`

If I customise the name, I'd expect that to change the affinity rules automatically - at least by default.

Reply:
From running some tests this looks to be fine.  If I do a 
```
helm template kpop ./karpenter
...
  template:
    metadata:
      labels:
        app.kubernetes.io/name: karpenter
        app.kubernetes.io/instance: kpop
...
```
The above are the labels for the karpenter deployment.  The app name is constant at karpenter while the instance name will change to whatever the release name is.

Reply:
What happens if you set `nameOverride` explicitly?

Reply:
Ok, in that instance it will change the name.  I'll see what can be done here.  Helm-fu is not my normal specialty.

Reply:
This was way harder than I would imagine to get the helm to do what is required.  Regardless, I learned something new today.